corehq/apps/locations/permissions.py

```diff
@@ -20,34 +21,57 @@ def is_locations_admin(view_fn):
     return locations_access_required(domain_admin_required(view_fn))
 
 
-def user_can_edit_location(user, location, project):
-    """
-    Expects SQLLocation
-    """
-    if not project.location_restriction_for_users:
+@quickcache(['user._id', 'project.name'])
```

Comment:
lol when i looked at quickcache's docs it said default was 10 seconds. Now I see there's a difference between timeout and memoize_timeout.

Anyways I didn't want to make crazy amounts of requests to psql from the API, so I used caching here. Should be able to be solved by using timeout=10, I believe.